### PR TITLE
feat(ui): project name formatting

### LINKED ISSUE #529: Sub in hyphens for spaces instead of showing error
Instead of saying the name is invalid we should automatically substitute a hyphen when the user types a space character.

image

### PR DESCRIPTION
**Pull request by sarahscott**
Replaces spaces with dashes for new project names; unit-tests name formatting

Closes #529

### CODE REVIEW

**Review by NunoAlexandre — COMMENTED** [review]
just two small comments :field_hockey:

**Review by xla — COMMENTED** [review]
Nothing to add to @NunoAlexandre observations, happy to approve once addressed. Test needs to be looked at.

**Comment by rudolfs** [comment]
I'm wondering whether we should also add this to the handle input in onboarding to keep things consistent, as it also disallows spaces. \cc @brandonhaslegs @juliendonck 

Screenshot 2020-09-21 at 09 53 43

**Comment by juliendonck** [comment]
> I'm wondering whether we should also add this to the handle input in onboarding as well to keep things consistent. That input also disallows spaces. \cc @brandonhaslegs @juliendonck
> 
> Screenshot 2020-09-21 at 09 53 43

yes :)

File: ui/src/project.ts [hunk 1]
```diff
@@ -227,6 +227,10 @@ export const defaultBranch = writable<string>(DEFAULT_BRANCH_FOR_NEW_PROJECTS);
 
 const projectNameMatch = "^[a-z0-9][a-z0-9._-]+$";
 
+export const formatNameInput = (input: string) => input.replace(" ", "-");
+export const formatFromExistingRepo = (path: string) =>
```
[the remark this answers is not in the document]
Comment (FintanH) [reply ui/src/project.ts:231]: And better still, you should represent the newly extracted name with a type that says you did so. That way you can't mix it up with other strings :grin:

[the remark this answers is not in the document]
Comment (xla) [reply ui/src/project.ts:231]: type police

[the remark this answers is not in the document]
Comment (FintanH) [reply ui/src/project.ts:231]: Put a type on it meow

[the remark this answers is not in the document]
Comment (sarahscott) [reply ui/src/project.ts:231]: Would be cool but TypeScript would actually just treat this as a `string` and wouldn't throw an error if I treated it as any other "type" of this form: 
```
type RepoPath = string;
```

There are some interesting approaches to `newtypes` in TS though - could be interesting to explore:
https://www.everythingfrontend.com/posts/newtype-in-typescript.html
https://github.com/gcanti/newtype-ts

File: cypress/integration/project_creation.spec.js [hunk 2]
```diff
@@ -130,18 +130,17 @@ context("project creation", () => {
       cy.get("body").type("{esc}", { force: true });
     });
 
-    context("name", () => {
+    context.only("name", () => {
```
Comment (rudolfs) [inline cypress/integration/project_creation.spec.js:133]:
```suggestion
    context("name", () => {
```

Our eslint plugin has failed us :( This should have failed on CI.

**Review by rudolfs — CHANGES_REQUESTED** [review]

File: ui/src/project.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (FintanH) [reply ui/src/project.ts]: We could have a class with a single field, no?

**Comment by rudolfs** [comment]
Other than the skipped tests works and looks great!

**Review by rudolfs — APPROVED** [review]

**Review by MeBrei — APPROVED** [review]

**Pull request merged**